src/lift.jl

```diff
@@ -0,0 +1,61 @@
+@inline function lift(f, x)
+    if null_safe_op(f, eltype(x))
+        return @compat Nullable(f(unsafe_get(x)), !isnull(x))
+    else
+        U = Core.Inference.return_type(f, Tuple{eltype(x)})
```

Comment:
We should probably check if `U` is leaf, yes? If so, we should decide what to do when `U` is abstract. Return `Union{}` as in 16961?